pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -308,7 +310,34 @@ func (r *ReconcileClusterDeployment) Reconcile(request reconcile.Request) (recon
 		cdLog.Errorf("error getting job: %v", err)
 		return reconcile.Result{}, err
 	} else {
+		// If install job exists but the cd generation is different, recreate the job after deleting it
+		// provided the cd is not installed
 		cdLog.Infof("cluster job exists, successful: %v", cd.Status.Installed)
+		if !cd.Status.Installed {
+			cdGeneration := strconv.FormatInt(cd.Generation, 10)
+			cdLog.Infof("cluster generation: %v", cdGeneration)
+			if existingJob.Annotations != nil {
+				if generation, ok := existingJob.Annotations[clusterDeploymentGeneration]; ok {
+					if generation != cdGeneration {
```

Comment:
instead of doing a string compare, do a number compare and check that the current generation is greater than the one that was used to create the job.